File: backend/src/main/java/com/ecom/salezone/services/AddressService.java
package com.ecom.salezone.services;

import com.ecom.salezone.dtos.AddAddressRequest;
import com.ecom.salezone.dtos.AddressDto;
import com.ecom.salezone.dtos.UpdateAddressRequest;

import java.util.List;

public interface AddressService {

    AddressDto addAddress(String userId, AddAddressRequest request, String logkey);

    List<AddressDto> getUserAddresses(String userId, String logkey);

    void deleteAddress(String userId, String addressId, String logkey);

    AddressDto setDefaultAddress(String userId, String addressId, String logkey);

    AddressDto getDefaultAddress(String userId, String logkey);

    AddressDto updateAddress(String userId, String addressId, UpdateAddressRequest request, String logkey);
}

File: frontend/src/services/addressService.js
import api from "./api";

const addressService = {
  getAddresses: async (userId) => {
    const res = await api.get(`/address/${userId}`);
    return res.data;
  },

  getDefaultAddress: async (userId) => {
    const res = await api.get(`/address/${userId}/default`);
    return res.data;
  },

  addAddress: async (userId, data) => {
    const res = await api.post(`/address/${userId}`, data);
    return res.data;
  },

  updateAddress: async (userId, addressId, data) => {
    const res = await api.put(`/address/${userId}/${addressId}`, data);
    return res.data;
  },

  deleteAddress: async (userId, addressId) => {
    const res = await api.delete(`/address/${userId}/${addressId}`);
    return res.data;
  },

  setDefault: async (userId, addressId) => {
    const res = await api.put(`/address/${userId}/default/${addressId}`);
    return res.data;
  },
};

export default addressService;
